[BUG] - Serialization rules aren't followed for arrays of nested objects

**Describe the bug**
If you have an array of nested objects, the `@SerializeProperty` rules for those objects are not followed.

**To Reproduce**
Here's at test I added to serialize_property_tests.ts:
```ts
test({
  name: "Serialize arrays of nested objects",
  fn() {
    class Test1 extends Serializable {
      @SerializeProperty("nested_property")
      serializeMe = 999;
    }
    class Test2 extends Serializable {
      @SerializeProperty({
        serializedKey: "outer_property",
      })
      nested: Test1[] = [new Test1()];
    }
    const testObj = new Test2();

    assertEquals(
      testObj.toJson(),
      `{"outer_property":[{"nested_property":999}}]`,
    );
  },
});
```

Output:
```
failures:

Serialize arrays of nested objects
AssertionError: Values are not equal:


    [Diff] Actual / Expected


-   "{\"outer_property\":[{\"serializeMe\":999}]}"
+   "{\"outer_property\":[{\"nested_property\":999}}]"
```

**Expected behaviour**
If you have an array of objects with serialization information, it should be used in toJson() / toPojo().

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - deno@1.5.1
 - std@0.76.0
 - fmt changes with deno upgrade
+- fixed #62
 
 ### Added
 - interface `tsTransformKey`

--- a/serializable.ts
+++ b/serializable.ts
@@ -127,7 +127,12 @@ export function toPojo(
     }
 
     if (Array.isArray(value)) {
-      record[serializedKey] = value.map((v: any) => toJsonStrategy(v));
+      record[serializedKey] = value.map((item: any) => {
+        if (item instanceof Serializable) {
+          return toPojo(item);
+        }
+        return toJsonStrategy(item);
+      });
     } else if (value !== undefined) {
       record[serializedKey] = toJsonStrategy(value);
     }

--- a/serialize_property_test.ts
+++ b/serialize_property_test.ts
@@ -380,3 +380,32 @@ test({
     );
   },
 });
+
+test({
+  name: "Serialize arrays of nested objects",
+  fn() {
+    class Test1 extends Serializable {
+      @SerializeProperty("nested_property")
+      serializeMe = 999;
+    }
+    class Test2 extends Serializable {
+      @SerializeProperty({
+        serializedKey: "outer_property",
+      })
+      nested: Test1[] = [new Test1()];
+    }
+
+    class Test3 extends Serializable {
+      @SerializeProperty({
+        serializedKey: "outer_outer_property",
+      })
+      nested2: Test2[] = [new Test2()];
+    }
+    const testObj = new Test3();
+
+    assertEquals(
+      testObj.toJson(),
+      `{"outer_outer_property":[{"outer_property":[{"nested_property":999}]}]}`,
+    );
+  },
+});

